GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/NotifyUsersAndRetireInactiveAccountsJob.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.Extensions;
+using GenderPayGap.WebUI.Services;
+
+namespace GenderPayGap.WebUI.BackgroundJobs.ScheduledJobs
+{
+    public class NotifyUsersAndRetireInactiveAccountsJob
+    {
+        private readonly IDataRepository dataRepository;
+        private readonly EmailSendingService emailSendingService;
+
+        public NotifyUsersAndRetireInactiveAccountsJob(IDataRepository dataRepository, EmailSendingService emailSendingService)
+        {
+            this.dataRepository = dataRepository;
+            this.emailSendingService = emailSendingService;
+        }
+
+        public void NotifyUsersAndRetireInactiveAccounts()
+        {
+            var runId = JobHelpers.CreateRunId();
+            var startTime = VirtualDateTime.Now;
+            JobHelpers.LogFunctionStart(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+
+            try
+            {
+                SendReminderEmails();
+
+                RetireUsers();
+
+                JobHelpers.LogFunctionEnd(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+            }
+            catch (Exception ex)
+            {
+                JobHelpers.LogFunctionError(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime, ex);
+
+                //Rethrow the error
+                throw;
+            }
+        }
+
+        private void SendReminderEmails()
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+            DateTime threeYearsAgoMinusThirtyDays = threeYearsAgo.AddDays(30);
+            DateTime threeYearsAgoMinusSevenDays = threeYearsAgo.AddDays(7);
+
+            List<User> usersToSendReminders = dataRepository.GetAll<User>()
+                .Where(u => InThirtyDayRange(u, threeYearsAgoMinusThirtyDays) || InSevenDayRange(u, threeYearsAgoMinusSevenDays) )

```

Comment:
I think this won't work, because of a limitation in Entity Framework (our ORM) - let's chat about this later

Basically, EF doesn't allow us call methods in `Where` statements